image/config.go

```diff
@@ -37,35 +35,29 @@ func findConfig(w walker, d *descriptor) (*config, error) {
 	var c config
 	cpath := filepath.Join("blobs", d.algo(), d.hash())
 
-	switch err := w.walk(func(path string, info os.FileInfo, r io.Reader) error {
-		if info.IsDir() || filepath.Clean(path) != cpath {
-			return nil
-		}
-		buf, err := ioutil.ReadAll(r)
-		if err != nil {
-			return errors.Wrapf(err, "%s: error reading config", path)
-		}
+	r, err := w.Get(*d)
+	if err != nil {
+		return nil, err
+	}
 
-		if err := schema.MediaTypeImageConfig.Validate(bytes.NewReader(buf)); err != nil {
-			return errors.Wrapf(err, "%s: config validation failed", path)
-		}
+	buf, err := ioutil.ReadAll(r)
+	if err != nil {
+		return nil, err
```

Comment:
one nit, I think it's better to keep the error message consistent, can we use 
`return nil, errors.Wrapf(err, "%s: error reading config", path)` here?